src/core/conversation.py

```diff
@@ -100,7 +111,31 @@ def perform_next_step(self) -> Message:
         else:
             raise Exception(f"Invalid conversation role: {last_message.role}")
 
-        return self._get_last_message()
+        last_message = self._get_last_message()
+        error_limit = 5
```

Comment:
Why don't we deduce the errors here from conversation messages? Currently we may generate incorrect code all the time and we will make 2,5x more API calls

Reply:
I will reduce limit to 3